pages/admin/dashboard.js

```diff
@@ -0,0 +1,229 @@
+import { useRouter } from "next/router";
+import { useEffect, useState } from "react";
+import { CircularProgress, Typography } from "@material-ui/core";
+import PropTypes, { array } from "prop-types";
+import styles from "../../styles/GameRow.module.css";
+import game from "../../helpers/game";
+
+import firebase from "../../utils/firebase";
+import AdminGameView from "./game";
+
+import Link from "next/link";
+import {
+  Box,
+  Button,
+  Card,
+  CardActions,
+  CardContent,
+  CardHeader,
+  Container,
+  IconButton,
+  Paper,
+  Snackbar,
+  TextField,
+} from "@material-ui/core";
+
+export default function Dashboard() {
+  const [gameArrayState, setGameArrayState] = useState([]);
+  const [gameClick, setGameClick] = useState(false);
+  const [login, setLogin] = useState(false);
+  const [gamePin, setGamePin] = useState();
+
+  const router = useRouter();
+
+  if (firebase.apps.length === 0) {
+    console.log("not initialized");
+  } else {
+    firebase.auth().onAuthStateChanged(function (user) {
+      if (user) {
+        setLogin(true);
+      } else {
+        //no user logged in, so go back to admin
+        setLogin(false);
+        router.replace(`/admin?login=signOut`);
+      }
+    });
+  }
+
+  useEffect(async () => {
+    var data;
+    const snapshot = await firebase
+      .database()
+      .ref(`games`)
+      .on("value", (subsnapshot) => {
+        data = subsnapshot.val();
+        //console.log(data);
+        setGameArrayState(data);
+      });
+  }, []);
+
+  var listOfGames = [];
+  for (var k in gameArrayState) {
+    listOfGames.push(k);
+  }
+
+  function gameClicked(pin) {
+    if (gameArrayState[pin].state !== "GAME_STATE-END") {
+      router.replace(`?gamepin=${pin}`);
+      setGameClick(true);
+      setGamePin(pin);
+    } else {
+      //show results
+      router.replace(`/admin/results?gamepin=${pin}`);
+    }
+  }
+
+  function dashMain() {
+    setGameClick(false);
+    setGamePin("");
+    router.replace(`/admin/dashboard`);
+  }
+
+  function shadowToggle(gamepin, uid) {
+    //document.getElementById(styles.username).style = 'background: red;';
+    firebase
+      .database()
+      .ref(`games/` + gamepin + "/users/" + uid)
+      .child("sban")
+      .set(
+        gameArrayState[gamepin].users[uid].sban !== undefined
+          ? !gameArrayState[gamepin].users[uid].sban
+          : true
+      );
+  }
```

Comment:
Also, maybe move this into `helpers/game.js` or an `helpers/admin.js` cause I see it's being repeated between files